### ISSUE TITLE
ps-bridge.ps1 fails to resolve in agent mode

### ISSUE THREAD

**Issue by roniyako** [GeoffWilliams/puppet-windows_firewall#5]
I changed the version for the module and the below is the error i get when i run this on Windows 2016, only thing changed was the module version.

Error: /Stage[main]/Windows_fw::Enablefw/Resources[windows_firewall_rule]: Failed to generate additional resources using 'generate': Execution of 'powershell.exe -ExecutionPolicy Bypass -File true show' returned 4294770688: Processing -File 'true' failed because the file does not have a '.ps1' extension. Specify a valid Windows PowerShell script file name, and then try again.
Windows PowerShell
Copyright (C) 2016 Microsoft Corporation. All rights reserved.
Info: Applying configuration version '*********************************************'
Error: Failed to apply catalog: Execution of 'powershell.exe -ExecutionPolicy Bypass -File true show' returned 4294770688: Processing -File 'true' failed because the file does not have a '.ps1' extension. Specify a valid Windows PowerShell script file name, and then try again.
Windows PowerShell
Copyright (C) 2016 Microsoft Corporation. All rights reserved.

**Comment by GeoffWilliams** [comment]
```
class windows_fw::enablefw (
   Hash[String, Hash]  $base_profiles  = {},
   Hash[String, Hash]  $profiles       = {},
   Hash[String, Hash]  $base_rules     = {},
   Hash[String, Hash]  $rules          = {},
   Hash[String, Hash]  $base_global    = {},
   Hash[String, Hash]  $global         = {},
   Hash[String, Hash]  $base_groups    = {},
   Hash[String, Hash]  $groups         = {},
   Boolean             $purge_rules    = false,
) {

 resources { "windows_firewall_rule":
   purge => $purge_rules,
 }

 deep_merge($base_profiles, $profiles).each |$key, $opts| {
   windows_firewall_profile { $key:
     * => $opts,
   }
 }

 deep_merge($base_rules, $rules).each |$key, $opts| {
   windows_firewall_rule {
     default:
       ensure => present,
     ;

     $key:
       * => $opts,
   }
 }

 deep_merge($base_groups, $groups).each |$key, $opts| {
   windows_firewall_group { $key:
     * => $opts,
   }
 }

 # only need shallow merge for global hash since its only one level deep
 windows_firewall_global { "global":
   * => merge($base_global, $global),
 }

}
```

**Comment by GeoffWilliams** [comment]
PE: 2017 3.5

**Comment by roniyako** [comment]
Full output of puppet run


C:\Windows\system32>puppet agent -t
Info: Using configured environment 'WindowsFW'
Info: Retrieving pluginfacts
Info: Retrieving plugin
Info: Retrieving locales
Info: Loading facts
Info: Caching catalog for servername
Error: /Stage[main]/Windows_fw::Enablefw/Resources[windows_firewall_rule]: Failed to generate additional resources using 'generate': Execution of 'powershell.exe -ExecutionPolicy Bypass -File true show' returned 4294770688: Processing -File 'true' failed because the file does not have a '.ps1' extension. Specify a valid Windows PowerShell script file name, and then try again.
Windows PowerShell
Copyright (C) 2016 Microsoft Corporation. All rights reserved.
Info: Applying configuration version 'puppetmaster-WindowsFW-d8784ff13da'
Error: Failed to apply catalog: Execution of 'powershell.exe -ExecutionPolicy Bypass -File true show' returned 4294770688: Processing -File 'true' failed because the file does not have a '.ps1' extension. Specify a valid Windows PowerShell script file name, and then try again.
Windows PowerShell
Copyright (C) 2016 Microsoft Corporation. All rights reserved.

**Comment by GeoffWilliams** [comment]
The bridge script filename is either wrong, missing or being lost

```
[0;36mDebug: Checking for ps-bridge.ps1 at C:/ProgramData/PuppetLabs/puppet/cache/lib/ps/windows_firewall/ps-bridge.ps1[0m
[0;36mDebug: Executing: 'powershell.exe -ExecutionPolicy Bypass -File true show'[0m
```

**Comment by canihavethisone** [comment]
Hey I was about to post this issue (experiencing it on Win10.1809 LTSC).  It also occurs when using 'puppet resource windows_firewall_rule'

Also, can your module accept a binding a firewall rule to a service?  I can't seem to get that to work and didn't see it in your readme.

**Comment by GeoffWilliams** [comment]
was a broken ternary due to bracket precedence - fix released in 0.2.2
